libavcodec/allcodecs.c

```diff
@@ -919,8 +943,17 @@ static const AVCodec *find_codec(enum AVCodecID id, int (*x)(const AVCodec *))
         if (p->id == id) {
             if (p->capabilities & AV_CODEC_CAP_EXPERIMENTAL && !experimental) {
                 experimental = p;
-            } else
+            }
+            else {
+#if defined(_WIN32) && CONFIG_MEDIAFOUNDATION
+                if (p->priv_class && p->priv_class->option &&
+                    stricmp(p->priv_class->option->name, "mfdec") == 0)
```

Comment:
This was a hack to find MF decoders on win platform builds.  I thought about making this a conditional compile option in the codec list itself, with MF versions or FFMpeg versions depending on the platform "_WIN32" macro definition, but went with this instead.  Bill later advised that it will be fine to simple compile the list of codecs with conditionals.   I will get the commit ready to clean this part up so this will go away completely.